### PR TITLE
Kubernetes 1.7.1

### LINKED ISSUE #1034: Kubernetes v1.7.1
## Acceptance Criteria
- Users can install Kubernetes v1.7.0 on Linux

### PR DESCRIPTION
**Pull request by lachie83**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1020)

Fixes #1034

### CODE REVIEW

File: pkg/acsengine/engine_test.go [hunk 1]
```diff
@@ -198,19 +198,23 @@ func addTestCertificateProfile(api *api.CertificateProfile) {
 
 func TestVersionOrdinal(t *testing.T) {
 	RegisterTestingT(t)
+	v171 := api.OrchestratorVersion("1.7.1")
```
Comment (seanknox) [inline pkg/acsengine/engine_test.go]: 🎉

File: examples/kubernetesversions/kubernetes1.7.1.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "apiVersion": "vlabs",
+  "properties": {
+    "orchestratorProfile": {
+      "orchestratorType": "Kubernetes",
+      "orchestratorVersion": "1.7.1"
+    },
```
Comment (seanknox) [inline examples/kubernetesversions/kubernetes1.7.1.json]: I want to start removing cluster configs that provide little value—I'd opt to delete this one.

File: pkg/acsengine/const.go [hunk 3]
```diff
@@ -80,6 +80,28 @@ const (
...
+		"hyperkube":       "hyperkube-amd64:v1.7.1",
+		"dashboard":       "kubernetes-dashboard-amd64:v1.6.1",
+		"exechealthz":     "exechealthz-amd64:1.2",
+		"addonresizer":    "addon-resizer:1.7",
+		"heapster":        "heapster:v1.3.0",
+		"dns":             "k8s-dns-kube-dns-amd64:1.14.4",
+		"addonmanager":    "kube-addon-manager-amd64:v6.4-beta.2",
+		"dnsmasq":         "k8s-dns-dnsmasq-amd64:1.13.0",
+		"pause":           "pause-amd64:3.0",
+		"windowszip":      "v1.7.1intwinnat.zip",
```
Comment (seanknox) [inline pkg/acsengine/const.go]: @JiangtianLi let's build up a Windows Kubernetes v1.7.1 zip. Windows support doesn't need to block this—we can roll out Linux support first.

File: pkg/acsengine/engine_test.go (hunk 1 above)
Comment (seanknox) [inline pkg/acsengine/engine_test.go]: 🎉  testing!

File: pkg/acsengine/engine_test.go [hunk 4]
```diff
@@ -198,19 +198,23 @@ func addTestCertificateProfile(api *api.CertificateProfile) {
...
 	v166 := api.OrchestratorVersion("1.6.6")
 	v162 := api.OrchestratorVersion("1.6.2")
 	v160 := api.OrchestratorVersion("1.6.0")
 	v153 := api.OrchestratorVersion("1.5.3")
 	v16 := api.OrchestratorVersion("1.6")
 
+	Expect(v171 > v166).To(BeTrue())
```
Comment (seanknox) [inline pkg/acsengine/engine_test.go]: On second glance, I'm not sure what these tests actually accomplish. Not a blocker for this PR.

**Comment by seanknox** [comment]
@JiangtianLi can you note in here when Windows v1.7.1 builds are available? @lachie83 not a blocker for this PR.

**Comment by JiangtianLi** [comment]
@seanknox Sure, will update here.

**Comment by seanknox** [comment]
@lachie83 go ahead and merge after CI passes, whenever you're ready.

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by marc-sensenich** [comment]
@lachie83 I tried deploying this today for testing and I ran into an issue with heapster-nanny container. It seems that in `addon-resizer:2.0` `/pod_nanny` has removed some of the flags using in the heapster deployment; e.g. `threshold` and `estimator`. which prevented the heapster pod to reach a running status

**Comment by seanknox** [comment]
giphy 2

**Pull request merged**